lib/tests/csrfProtected.test.js

```diff
@@ -0,0 +1,149 @@
+import mockedAxios from "axios";
+import { createMocks } from "node-mocks-http";
+import { submitToAPI } from "../integration/helpers";
+import { logMessage } from "@lib/logger";
+import TestForm from "./testData";
+import { getCsrfToken } from "next-auth/client";
+import { csrfProtected } from "@lib/middleware/csrfProtected";
+
+jest.mock("axios");
+jest.mock("next-auth/client");
+
+const mockResponses = {
+  1: "test response", // TextField
+  2: "test response 2", // TextArea
+  4: "Manitoba", // DropDown
+  5: "Student", // Radio
+  6: [], // CheckBox
+  7: [
+    {
+      0: "Doe",
+      1: "John",
+      2: "CDS",
+      3: ["No"],
+    },
+    {
+      0: "Seat",
+      1: "Love",
+      2: "CHMC",
+      3: ["Yes", "Other"],
+    },
+  ],
+  8: "",
+};
+
+describe("Csrf Protection middleware", () => {
+  let mockedConsoleError;
+  beforeEach(() => {
+    mockedConsoleError = jest.spyOn(logMessage, "error");
+    mockedConsoleError.mockImplementation(() => {});
+  });
+  afterEach(() => {
+    mockedConsoleError.mockRestore();
+    getCsrfToken.mockRestore();
+  });
+
+  test("Should pass with unprotected method GET", async () => {
+    const { req, res } = createMocks({
+      method: "GET",
+    });
+    jest.spyOn(logMessage, "error");
+
+    await csrfProtected(["POST", "PUT"])(req, res);
+    expect(res.statusCode).toBe(200);
+  });
+
+  test("Should fail with protected method POST and a different csrf token", async () => {
+    getCsrfToken.mockReturnValueOnce("differentCsrfToken");
+    const { req, res } = createMocks({
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "x-csrf-token": "csrfToken",
+      },
+    });
+    jest.spyOn(logMessage, "error");
+
+    await csrfProtected(["POST"])(req, res);
+    expect(res.statusCode).toBe(403);
+    expect(JSON.parse(res._getData())).toEqual({
+      error: "Access Denied",
+    });
+  });
+
+  test("Should fail with null csrf token", async () => {
+    getCsrfToken.mockReturnValueOnce(null);
+    const { req, res } = createMocks({
+      method: "POST",
+    });
+    jest.spyOn(logMessage, "error");
+
+    await csrfProtected(["POST"])(req, res);
+    expect(res.statusCode).toBe(403);
+    expect(JSON.parse(res._getData())).toEqual({
+      error: "Access Denied",
+    });
+  });
+
+  test("Should allow the request with valid method and Csrf Token", async () => {
+    getCsrfToken.mockReturnValueOnce("csrfToken");
+    const { req, res } = createMocks({
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "x-csrf-token": "csrfToken",
+      },
+    });
+    jest.spyOn(logMessage, "error");
+
+    await csrfProtected(["POST"])(req, res);
+    expect(res.statusCode).toBe(200);
+  });
+});
+
+describe("Submit a form with csrf protection enable", () => {
+  let mockedConsoleError;
```

Comment:
Since logMessage.error was removed from the csrfProtected function we should also remove the jest spy on the function from the tests.